monitoring/monitoring_internal_test.go

```diff
@@ -181,528 +192,543 @@ func TestPeriodicReport(t *testing.T) {
-					MinThreshold: 0,
-					MaxThreshold: 0,
-				},
-				InTraffic: &config.AlertRule{
-					MinTimeout:   config.Duration{},
-					MinThreshold: 0,
-					MaxThreshold: 0,
-				},
-				OutTraffic: &config.AlertRule{
-					MinTimeout:   config.Duration{},
-					MinThreshold: 0,
-					MaxThreshold: 0,
-				},
+	var expectedInstanceAlertCount int
+
+	monitoringInstances := []cloudprotocol.InstanceMonitoringData{
+		{
+			InstanceIdent: cloudprotocol.InstanceIdent{
+				ServiceID: "service1",
+				SubjectID: "subject1",
+				Instance:  1,
+			},
+			RAM:        1100,
+			CPU:        uint64(math.Round(35 / float64(runtime.NumCPU()))),
+			UsedDisk:   2300,
+			InTraffic:  150,
+			OutTraffic: 150,
+		},
+		{
+			InstanceIdent: cloudprotocol.InstanceIdent{
+				ServiceID: "service2",
+				SubjectID: "subject1",
+				Instance:  1,
+			},
+			RAM:        2100,
+			CPU:        uint64(math.Round(45 / float64(runtime.NumCPU()))),
+			UsedDisk:   2300,
+			InTraffic:  250,
+			OutTraffic: 150,
+		},
+		{
+			InstanceIdent: cloudprotocol.InstanceIdent{
+				ServiceID: "service1",
+				SubjectID: "subject2",
+				Instance:  2,
 			},
-		}); err != nil {
-		t.Fatalf("Can't start monitoring service: %s", err)
+			RAM:        2200,
+			CPU:        uint64(math.Round((45 + 45) / float64(runtime.NumCPU()))),
+			UsedDisk:   4600,
+			InTraffic:  150,
+			OutTraffic: 250,
+		},
+		{
+			InstanceIdent: cloudprotocol.InstanceIdent{
+				ServiceID: "service1",
+				SubjectID: "subject2",
+				Instance:  2,
+			},
+			RAM:        2200,
+			CPU:        uint64(math.Round((45 + 45) / float64(runtime.NumCPU()))),
+			UsedDisk:   2300,
+			InTraffic:  150,
+			OutTraffic: 250,
+		},
 	}
 
-	terminate := false
-
-	for terminate != true {
-		select {
-		case data := <-monitor.GetMonitoringDataChannel():
-			if len(data.ServiceMonitoring) != 2 {
-				t.Errorf("Wrong number of services: %d", len(data.ServiceMonitoring))
-			}
+	for i, item := range testData {
+		processesData = append(processesData, &testProcessData{
+			uid:       int32(item.monitoringConfig.UID),
+			quotaData: item.quotaData,
+		})
 
-			for resource, numAlerts := range alertMap {
-				if numAlerts != 2 {
-					t.Errorf("Wrong number of %s alerts: %d", resource, numAlerts)
-				}
-			}
+		instanceID := fmt.Sprintf("instance%d", i)
+		if err := monitor.StartInstanceMonitor(instanceID, item.monitoringConfig); err != nil {
+			t.Fatalf("Can't start monitoring service: %s", err)
+		}
 
-			if len(alertMap) != 5 {
-				t.Error("Not enough alerts")
-			}
+		instanceTrafficMonitoringData[instanceID] = item.trafficMonitoring
+		expectedInstanceAlertCount += item.alertExpectedCount
+	}
 
-			terminate = true
+	defer func() {
+		processesData = nil
+	}()
 
-		case <-time.After(sendDuration * 2):
-			t.Fatal("Monitoring data timeout")
+	select {
+	case monitoringData := <-monitor.GetMonitoringDataChannel():
+		if len(alertMap) != expectedInstanceAlertCount {
+			t.Error("Alerts should not be generated")
```

Comment:
Should be fatal, and message should be fixed.

Reply:
Done